pkgs/development/python-modules/mlx-lm/default.nix

```diff
@@ -0,0 +1,78 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  wheel,
+  jinja2,
+  mlx,
+  numpy,
+  protobuf,
+  pyyaml,
+  transformers,
+  sentencepiece,
+  pytestCheckHook,
+  writableTmpDirAsHomeHook,
+}:
+
+buildPythonPackage rec {
+  pname = "mlx-lm";
+  version = "0.26.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ml-explore";
+    repo = "mlx-lm";
+    tag = "v${version}";
+    hash = "sha256-J69XIqsjQ4sQqhx+EkjKcVXVlQ4A4PGJvICSiCfoSOA=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    jinja2
+    mlx
+    numpy
+    protobuf
+    pyyaml
+    transformers
+  ];
+
+  nativeCheckInputs = [
+    writableTmpDirAsHomeHook
+    pytestCheckHook
+    sentencepiece
+  ];
+
+  pythonImportsCheck = [
+    "mlx_lm"
+  ];
+
+  disabledTestPaths = [
+    # Requires network access to huggingface.co
+    "tests/test_datsets.py"
+    "tests/test_generate.py"
+    "tests/test_server.py"
+    "tests/test_tokenizers.py"
+    "tests/test_utils.py::TestUtils::test_convert"
+    "tests/test_utils.py::TestUtils::test_load"
```

Comment:
I'm not sure the `::...` part is correctly handled by `disabledTestPaths`. You may use `disabledTests` instead, or `--deselect=...` in `pytestFlagsArray`.

Reply:
@GaetanLepage it's handled and is useful when the same test name is used in multiple places.

Reply:
Ok, I thought only actual file paths could be provided to `disabledTestPaths`. TIL!